File: edukate-backend/src/main/java/io/github/sanyavertolet/edukate/backend/dtos/CreateBundleRequest.java
package io.github.sanyavertolet.edukate.backend.dtos;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
public class CreateBundleRequest {
    @NotBlank
    @Size(max = 50)
    private String name;
    @NotBlank
    @Size(max = 255)
    private String description;
    @NotNull
    private Boolean isPublic;
    @NotEmpty
    private List<@NotBlank String> problemIds;
}

File: edukate-backend/src/main/kotlin/io/github/sanyavertolet/edukate/backend/dtos/CreateBundleRequest.kt
package io.github.sanyavertolet.edukate.backend.dtos

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.Size

data class CreateBundleRequest(
    @field:NotBlank @field:Size(max = 50) val name: String,
    @field:NotBlank @field:Size(max = 255) val description: String,
    val isPublic: Boolean,
    @field:NotEmpty val problemIds: List<@NotBlank String>,
)
